server/src/main/java/org/elasticsearch/action/bulk/BulkPrimaryExecutionContext.java

```diff
@@ -184,6 +184,7 @@ public void resetForExecutionForRetry() {
         currentItemState = ItemProcessingState.INITIAL;
         requestToExecute = null;
         executionResult = null;
+        retryCounter++;
```

Comment:
Oh, I was actually hoping to be able to use the retry counter for mapping updates in this PR: https://github.com/elastic/elasticsearch/pull/96235, which re-tries failed mapping updates once.

Reply:
But I guess it makes sense to have two distinct counters then. Otherwise, if the update results in a mapping update, it may "steal" a retry that was suppose to be used for retrying the update.

Reply:
Yes, that would be odd i think.

Reply:
I've updated the variable and methods to `updateRetryCounter` to make it clear that this counter's scope is specific to update operations.

Reply:
> It would be good to amend one of the mapping update retry tests to catch this first if we can. Probably testExecuteBulkIndexRequestWithMappingUpdates or in BulkWithUpdatesIT.testBulkUpdateWithScriptedUpsertAndDynamicMappingUpdate (set retryOnConflict=1 randomly there).

Did you attempt this and found it infeasible? I think it is worth ensuring that these are indeed independent now.

Reply:
I've amended `testExecuteBulkIndexRequestWithMappingUpdates` as you suggested:

https://github.com/elastic/elasticsearch/pull/96262/files#diff-55895680ebb0a3c7cbf2b86c9c71fb8f98eeae939f001250194223e02aaa533fR280

Reply:
Thanks, that is probably enough.